lib/language.js

```diff
@@ -7,42 +7,38 @@
 const locale = require('locale')
 
 const cookieName = 'language'
+const cookieOptions = { secure: true, sameSite: 'strict', httpOnly: true }
 const validLanguages = ['sv', 'en']
 const defaultLanguage = validLanguages[0]
 const supportedLocales = new locale.Locales(validLanguages, defaultLanguage)
 
+const validLanguage = lang => {
+  if (validLanguages.includes(lang)) {
+    return lang
+  }
+}
+
 /**
  * Initialize locale and language for the current user session (respects cookie: language).
  */
 function _init(req, res, newLang) {
-  let lang = newLang || req.cookies[cookieName]
-
-  // Only allow lang to be one of the valid languages
-  if (lang && !validLanguages.includes(lang)) lang = defaultLanguage
+  let lang =
+    validLanguage(newLang) ||
+    validLanguage(req.cookies[cookieName]) ||
+    validLanguage(req.headers['accept-language']) ||
+    defaultLanguage
 
-  let chosenLocale
+  const locales = new locale.Locales(lang, defaultLanguage)
+  const chosenLocale = locales.best(supportedLocales)
 
-  if (lang) {
-    const locales = new locale.Locales(lang, defaultLanguage)
-    chosenLocale = locales.best(supportedLocales)
-  } else {
-    const locales = new locale.Locales(req.headers['accept-language'], defaultLanguage)
-    chosenLocale = locales.best(supportedLocales)
-  }
-
-  // If we got an explicit language we set the language cookie
-  if (newLang) {
-    res.cookie(cookieName, chosenLocale.language, { secure: true, sameSite: 'strict', httpOnly: true })
-  } else if (!req.cookies[cookieName]) {
-    // Make sure language cookie is set so subsequent requests are guaranteed to use the same language
-    res.cookie(cookieName, lang || 'sv', { secure: true, sameSite: 'strict', httpOnly: true })
+  // Don't reset the cookie with the same language
```

Comment:
kill this comment and put it in a function name if there is need to explain

Reply:
Killed it.